plugins/@grouparoo/zendesk/src/lib/export/exportProfile.ts

```diff
@@ -0,0 +1,253 @@
+import { ExportProfilePluginMethod } from "@grouparoo/core";
+import { connect } from "../connect";
+import {
+  getBuiltInFields,
+  getRequiredFields,
+  getUserFields,
+} from "./destinationMappingOptions";
+
+export const exportProfile: ExportProfilePluginMethod = async ({
+  appOptions,
+  export: {
+    toDelete,
+    newProfileProperties,
+    oldProfileProperties,
+    newGroups,
+    oldGroups,
+  },
+}) => {
+  try {
+    const client = await connect(appOptions);
+    let external_id = newProfileProperties.external_id;
+    let email = newProfileProperties.email;
+
+    const dateValue = newProfileProperties.date_field;
+
+    // console.log("exportProfile", newProfileProperties);
+
+    if (!external_id) {
+      throw new Error(
+        `newProfileProperties[external_id] is a required mapping`
+      );
+    }
+
+    // zendesk does this itself
+    if (email) {
+      email = email.toLowerCase().trim();
+    }
+
+    const user = await findUser(
+      client,
+      newProfileProperties,
+      oldProfileProperties
+    );
+
+    if (toDelete) {
+      if (user) {
+        // this does a soft delete
+        await client.users.delete(user.id);
+
+        // TODO: should we fully delete it? (see test for how to set it up)
```

Comment:
I think you made the right choice with soft-delete.  I can't think of a reason to ever /really/ delete a Zendesk profile unless there's a request to do so by the customer.